src/dune_rules/modules.ml

```diff
@@ -447,75 +443,148 @@ module Group = struct
+      | None ->
+        Code_error.raise "for_alias: not found"
+          [ ("alias", Module.to_dyn alias) ]
+
+    let path_of_alias_module alias =
       match Module.kind alias with
       | Alias for_ -> for_
       | _ -> Code_error.raise "for_alias: not an alias module" []
-    in
-    let rec loop (t : t) = function
-      | [] -> t
-      | name :: path -> (
-        match Module_name.Map.find t.modules name with
-        | Some (Group g) -> loop g path
-        | Some (Module m) ->
-          Code_error.raise "for_alias: unexpected module"
-            [ ("m", Module.to_dyn m); ("alias", Module.to_dyn alias) ]
-        | None ->
-          Code_error.raise "for_alias: not found"
-            [ ("alias", Module.to_dyn alias) ])
-    in
-    let group = loop t path in
-    let modules =
-      match group.name with
-      | None -> group.modules
-      | Some name -> Module_name.Map.remove group.modules name
-    in
-    Module_name.Map.filter_map modules ~f:(fun (g : node) ->
-        match g with
-        | Module m -> Some m
-        | Group g -> lib_interface g)
+
+    let make_for_alias t alias path =
+      let rec loop (t : t) = function
+        | [] -> t
+        | name :: path ->
+          let group = find_module alias t.modules name in
+          loop group path
+      in
+      let group = loop t path in
+      let modules = Module_name.Map.remove group.modules group.name in
+      Module_name.Map.filter_map modules ~f:(fun (g : node) ->
+          match g with
+          | Module m -> Some m
+          | Group g -> lib_interface g)
+  end
+
+  let for_alias t alias =
+    For_alias.make_for_alias t alias (For_alias.path_of_alias_module alias)
 end
 
 module Unwrapped = struct
+  type t = Group.node Module_name.Map.t
+
   let decode ~src_dir =
     let open Dune_lang.Decoder in
-    Group.decode ~src_dir
+    Group.decode_modules ~src_dir
     <|> let+ map = Module.Name_map.decode ~src_dir in
-        { Group.alias = None
-        ; name = None
-        ; modules = Module_name.Map.map map ~f:(fun m -> Group.Module m)
-        }
+        Module_name.Map.map map ~f:(fun m -> Group.Module m)
+
+  let encode = Group.encode_modules
+
+  let to_dyn t = Module_name.Map.to_dyn Group.dyn_of_node t
+
+  let of_trie trie ~mangle ~src_dir =
+    Module_name.Map.mapi trie ~f:(fun name (m : 'a Module_trie.node) ->
+        match m with
+        | Map trie ->
+          let group =
+            Group.Of_trie.of_trie trie ~mangle ~src_dir ~interface:name
+              ~rev_path:[ name ]
+          in
+          Group.Group group
+        | Leaf m ->
+          let m = Module.set_path m [ name ] in
+          Module m)
+
+  let find (t : t) name =
+    match Module_name.Map.find t name with
+    | Some (Module m) -> Some m
+    | Some (Group _) | None -> None
 
-  let encode t = Group.encode t
+  let parents t m = Group.parents_modules [] t m
+
+  let find_dep t ~of_ name =
+    match Module.kind of_ with
+    | Alias _ -> Ok []
```

Comment:
is this ever the case in `(wrapped false)`? I thought there was no alias module.

Reply:
There are alias modules in `(wrapped false)` whenever `(include_subdirs qualified)` is enabled for modules in sub directories. There's just no alias module in the top level.